edgelet/edgelet-docker/src/runtime.rs

```diff
@@ -593,19 +602,17 @@ mod tests {
 
     #[test]
     fn merge_env_empty() {
-        let cur_env = Some(vec![]);
+        let cur_env = Some(&[][..]);
         let new_env = HashMap::new();
-        assert_eq!(
-            0,
-            DockerModuleRuntime::merge_env(cur_env.as_ref(), &new_env).len()
-        );
+        assert_eq!(0, DockerModuleRuntime::merge_env(cur_env, &new_env).len());
     }
 
     #[test]
     fn merge_env_new_empty() {
         let cur_env = Some(vec!["k1=v1".to_string(), "k2=v2".to_string()]);
         let new_env = HashMap::new();
-        let mut merged_env = DockerModuleRuntime::merge_env(cur_env.as_ref(), &new_env);
+        let mut merged_env =
+            DockerModuleRuntime::merge_env(cur_env.as_ref().map(AsRef::as_ref), &new_env);
```

Comment:
Is it necessary to call `as_ref()` _AND_ `map(AsRef::as_ref)` on `cur_env` here?

Reply:
`cur_env` is `Option<Vec<String>>` and `merge_env` wants `Option<&[String]>`. `Option::as_ref` is `&Option<T> -> &Option<T>`. `AsRef::as_ref` is `&T -> &U`

Reply:
Got it, thanks